arelle/plugin/validate/ESEF/__init__.py

```diff
@@ -762,6 +778,10 @@ def checkFootnote(elt: ModelInlineFootnote | ModelResource, text: str) -> None:
                         modelXbrl.warning("ESEF.2.2.2.percentGreaterThan100",
                             _("A percent fact should have value <= 100: %(element)s in context %(context)s value %(value)s"),
                             modelObject=f, element=f.qname, context=f.context.id, value=f.xValue)
+                    elif f.concept.balance is not None and f.xValue < 0:
```

Comment:
This check should also probably contain an xValid check so that we do not run into weird instances where xValue is a string.

Reply:
The suggested check on xValid has already been done some lines higher:

https://github.com/Arelle/Arelle/blob/50a5692f74a2f4c64ae1186714bc3f7e762009fe/arelle/plugin/validate/ESEF/__init__.py#L774